src/api-v1.js

```diff
@@ -90,10 +91,12 @@ api.declare({
     let result;
 
     do {
+      let startTime = new Date();
```

Comment:
i like easy timing in the library, but in some cases i might have to create the times in a weird place and just want to set a start, calcuate the end and do the calcuation.  I wouldn't want to be limited to only a promise or function for duration calculations